docs/js-core-2/week-1/homework.md

```diff
@@ -4,6 +4,190 @@ title: Coursework
+
+console.log(
+  "In the last 5 days, the stock changed by",
+  stock.changeInLastFiveDays
+);
+```
+
+The output should be
+
+```
+In the last 5 days, the stock changed by -7
+```
+
+3.1 Write a program that prints out to the console information from the following object
+
+```js
+uk = {
+  name: "The United Kingdom of Great Britain and Northern Ireland",
+  shortName: "The UK",
+  capitalCity: {
+    name: "London",
+    population: 9648000,
+  },
+};
+```
+
+The output should be
+
+```
+The capital of The UK is London. London has a population of 9648000
+```
+
+3.2 Complete the following program so that it prints out information about the stocks that went down in the past 5 days.
+
+```js
+const stocks = [
+  {
+    name: "aapl",
+    lastClosingPrice: 172.99,
+    lastFiveDaysClosingPrices: [179.99, 180.33, 176.28, 175.64, 172.99],
+  },
+  {
+    name: "msft",
+    lastClosingPrice: 327.29,
+    lastFiveDaysClosingPrices: [340.69, 342.45, 334.69, 333.2, 327.29],
+  },
+  {
+    name: "amzn",
+    lastClosingPrice: 3408.34,
+    lastFiveDaysClosingPrices: [3384.44, 3393.39, 3421.37, 3420.74, 3408.34],
+  },
+  {
+    name: "googl",
+    lastClosingPrice: 2869.45,
+    lastFiveDaysClosingPrices: [2951.88, 2958.13, 2938.33, 2928.3, 2869.45],
+  },
+  {
+    name: "tsla",
+    lastClosingPrice: 938.53,
+    lastFiveDaysClosingPrices: [1101.3, 1093.94, 1067.0, 1008.87, 938.53],
+  },
+];
+
+function getMessage(stock) {
+  // return a message containing the name and last closing price of the stock
+}
+
+function isStockPriceDecreasing(stock) {
+  // return a boolean indicating whether the stock decreased in price over the last 5 days
+}
+
+const decreasingStocks = stocks.filter(isStockPriceDecreasing);
+const messages = decreasingStocks.map(getMessage);
+console.log("Stocks that decreased in price over the last 5 days:");
+for (const message of messages) {
+  console.log(message);
+}
+```
+
+Most of the code is already written. You should only modify the functions `isStockPriceDecreasing` and `getMessage`
+
+The output should be
+
+```
+Stocks that decreased in price over the last 5 days:
+aapl last closing at 172.99
+msft last closing at 327.29
+googl last closing at 2869.45
+tsla last closing at 938.53
+```
+
+4. Complete the `lastClosingPrice` method of the stock object so that it returns the last element of the `lastFiveDaysClosingPrices` array. This practice exercise is optional.
+
+```js
+let stock = {
+  name: "aapl",
+  lastClosingPrice: function () {
+    return this.lastFiveDaysClosingPrices[4];
+  },
+  lastFiveDaysClosingPrices: [179.99, 180.33, 176.28, 175.64, 173.99],
+};
+
+console.log(stock.lastClosingPrice());
+```
+
+The output should be
+
+```
+173.99
+```
+
+5. Add a test to your solution to `1-weather-report.js` in `JavaScript-Core-1-Coursework-Week3`. This test should check that when 3 city names are passed as an argument to the `getTemperatureReport` function, the returned array should be of length 3.
```

Comment:
5. Add a test to your solution to `1-weather-report.js` in `JavaScript-Core-1-Coursework-Week3`. 

When 3 city names are passed as an argument to the `getTemperatureReport` function, the returned array should be of length 3.